### PR TITLE
Support multi dimension values and configurable metricValues on AWS Cloudwatch Scaler

### PR DESCRIPTION
**Pull request by JonasMatos0**
…loudwatch Scaler

Co-authored-by: Jonas Matos <js.matos@sidi.org.br>
Co-authored-by: Thiago Teodoro Rodrigues  <t.teodoro@sidi.org.br>

_Provide a description of what has been changed_

### Checklist

- [x] Commits are signed with Developer Certificate of Origin (DCO)
- [x] Tests have been added
- [x] A PR is opened to update the documentation on https://github.com/kedacore/keda-docs
- [x] Changelog has been updated

Fixes #

### CODE REVIEW

**Comment by ThiagoTeodoro** [comment]
My name is wrong, pls when its possible make the correction =D , just to not forget.
Thanks for mentioned me as Co-authored.
Name : Thiago Teodoro Rodrigues 
@JonasMatos0

**Review by ahmelsayed — APPROVED** [review]

File: pkg/scalers/aws_cloudwatch_scaler.go [hunk 1]
```diff
@@ -66,9 +67,32 @@ func NewAwsCloudwatchScaler(config *ScalerConfig) (Scaler, error) {
...
+		}
+	} else {
+		meta.metricStatPeriod = defaultMetricStatPeriod
+	}
+
+	if val, ok := config.TriggerMetadata["metricStat"]; ok && val != "" {
+		meta.metricStat = val
+	} else {
+		meta.metricStat = defaultMetricStat
+	}
```
Comment (ahmelsayed) [inline pkg/scalers/aws_cloudwatch_scaler.go:95]: I think this is what the static check is complaining about with `cyclomatic complexity 35 of func `parseAwsCloudwatchMetadata` is high (> 30)`

I personally think it's fine, but maybe if you change it to

```go
	meta.metricCollectionTime = defaultMetricCollectionTime
	meta.metricStatPeriod = defaultMetricStatPeriod
	meta.metricStat = defaultMetricStat

	if val, ok := config.TriggerMetadata["metricCollectionTime"]; ok && val != "" {
		n, err := strconv.ParseInt(val, 10, 64)
		if err != nil {
			return nil, fmt.Errorf("metricCollectionTime not a valid number")
		}
		meta.metricCollectionTime = n
	}

	if val, ok := config.TriggerMetadata["metricStatPeriod"]; ok && val != "" {
		n, err := strconv.ParseInt(val, 10, 64);
		if err != nil {
			return nil, fmt.Errorf("metricStatPeriod not a valid number")
		} 
		meta.metricStatPeriod = n
	}

	if val, ok := config.TriggerMetadata["metricStat"]; ok && val != "" {
		meta.metricStat = val
	}
```

That should be equivalent to what you have but should reduce the cyclomatic complexity.

Comment (ThiagoTeodoro) [reply pkg/scalers/aws_cloudwatch_scaler.go:95]: We will try make the fill of 
`
        meta.metricCollectionTime = defaultMetricCollectionTime
	meta.metricStatPeriod = defaultMetricStatPeriod
	meta.metricStat = defaultMetricStat
`
In a function, maybe this avoid the problem with cyclomatic complexity, what you think @ahmelsayed

**Review by zroubalik — COMMENTED** [review]
Looking good, could you please open PR with this change in https://github.com/kedacore/keda-docs? And add this to the Changelog (Improvements section)

**Review by zroubalik — APPROVED** [review]
LGTM, thanks!

**Comment by zroubalik** [comment]
We can merge this one, once the docs PR si ready: https://github.com/kedacore/keda-docs/pull/352

**Pull request merged**